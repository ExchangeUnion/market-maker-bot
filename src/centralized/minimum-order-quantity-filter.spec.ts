import {
  quantityAboveMinimum,
  MINIMUM_ORDER_SIZE,
} from './minimum-order-quantity-filter';
import BigNumber from 'bignumber.js';

describe('quantityAboveMinimum', () => {
  it('returns true for minimum ETH quantity', () => {
    expect(quantityAboveMinimum('ETH')(MINIMUM_ORDER_SIZE.ETH)).toEqual(true);
  });

  it('returns false less than minimum ETH quantity', () => {
    const quantity = MINIMUM_ORDER_SIZE.ETH.minus(
      new BigNumber('0.000000000000000001')
    );
    expect(quantityAboveMinimum('ETH')(quantity)).toEqual(false);
  });

  it('returns true for minimum DAI quantity', () => {
    expect(quantityAboveMinimum('DAI')(MINIMUM_ORDER_SIZE.DAI)).toEqual(true);
  });

  it('returns false less than minimum DAI quantity', () => {
    const quantity = MINIMUM_ORDER_SIZE.DAI.minus(
      new BigNumber('0.000000000000000001')
    );
    expect(quantityAboveMinimum('DAI')(quantity)).toEqual(false);
  });

  it('returns true for minimum USDT quantity', () => {
    expect(quantityAboveMinimum('USDT')(MINIMUM_ORDER_SIZE.USDT)).toEqual(true);
  });

  it('returns false less than minimum USDT quantity', () => {
    const quantity = MINIMUM_ORDER_SIZE.USDT.minus(
      new BigNumber('0.000000000000000001')
    );
    expect(quantityAboveMinimum('USDT')(quantity)).toEqual(false);
  });

  it('throws for unknown asset KILRAU', () => {
    expect.assertions(1);
    const quantity = new BigNumber('1');
    expect(() => {
      quantityAboveMinimum('KILRAU')(quantity);
    }).toThrowErrorMatchingSnapshot();
  });
});
