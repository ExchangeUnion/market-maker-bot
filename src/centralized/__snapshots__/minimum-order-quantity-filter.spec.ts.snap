// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quantityAboveMinimum throws for unknown asset KILRAU 1`] = `"Could not retrieve minimum order quantity for KILRAU"`;
